### ISSUE TITLE
Azure reporting back that there is no edge zone in eastus

### ISSUE THREAD

**Issue by neozaga** [Azure/terraform-azurerm-avm-ptn-vnetgateway#22]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Greenfield/Brownfield provisioning

greenfield

### Terraform Version

ADO Pipeline

### Module Version

0.2.2

### AzureRM Provider Version

3.93.0

### Affected Resource(s)/Data Source(s)

module.avm-ptn-vnetgateway.azurerm_virtual_network_gateway.vgw

### Terraform Configuration Files

```hcl
module "avm-ptn-vnetgateway" {
  source  = "Azure/avm-ptn-vnetgateway/azurerm"
  #Version last updated 2024-02-14
  version = "0.2.2" # change this to your desired version, https://www.terraform.io/language/expressions/version-constraints

  # Begin required inputs
  location                            = azurerm_resource_group.rg_azure_central_networking_prod_eastus.location
  name                                = "vgw-azure-central-networking-prod-eastus"
  sku                                 = "VpnGw2AZ" # https://learn.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-about-vpngateways#gwsku
  subnet_address_prefix               = "" # The address prefix for the gateway subnet. Either subnet_id or subnet_address_prefix must be specified.
  type                                = "Vpn" # The type of the Virtual Network Gateway, ExpressRoute or VPN.
  virtual_network_name                = azurerm_virtual_network.vnet_azure_central_networking_prod_eastus.name
  virtual_network_resource_group_name = azurerm_resource_group.rg_azure_central_networking_prod_eastus.name
  # End required inputs

  # Begin optional inputs
  default_tags = {
      ApplicationName   = local.application_name
      Environment       = local.environment
      #CostCenter        = local.cost_center
      ProductOwnerEmail = local.product_owner_email
      AlternateEmail    = local.alternate_email
  }
  edge_zone = "1" #Error: Creating/Updating Virtual Network Gateway: (Name "vgw-azure-central-networking-prod-eastus" / Resource Group "rg-azure-central-networking-prod-eastus"): network.VirtualNetworkGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="UnsupportedEdgeZone" Message="Edge zone '1' not found. The available edge zones in location 'eastus' are ''."
  
 
  ip_configurations = {
      first_ip_configuration = { 
          ip_configuration_name = "ipc-vng-azure-central-networking-prod-eastus"
          apipa_addresses = []
          public_ip = {
              name = "pip-vng-azure-central-networking-prod-eastus"
              allocation_method = "Static"
              sku = "Standard" # https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-addresses#sku
          }
      }
  }
```

### tfvars variables values

```hcl
edge_zone = 1
```

### Debug Output/Panic Output

```shell
module.avm-ptn-vnetgateway.azurerm_virtual_network_gateway.vgw: Creating...
╷
│ Error: Creating/Updating Virtual Network Gateway: (Name "vgw-azure-central-networking-prod-eastus" / Resource Group "rg-azure-central-networking-prod-eastus"): network.VirtualNetworkGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="UnsupportedEdgeZone" Message="Edge zone '1' not found. The available edge zones in location 'eastus' are ''."
│ 
│   with module.avm-ptn-vnetgateway.azurerm_virtual_network_gateway.vgw,
│   on .terraform/modules/avm-ptn-vnetgateway/main.tf line 59, in resource "azurerm_virtual_network_gateway" "vgw":
│   59: resource "azurerm_virtual_network_gateway" "vgw" {
│ 
╵
```

### Expected Behaviour

There should be edge zones in eastus and we are using a vnetgateway that supports multiple AZs

### Actual Behaviour

Resource not created, terraform reporting that there are no edge zones in eastus

### Steps to Reproduce

terraform plan
terraform apply

**Comment by phx-tim-butters** [comment]
Hi There,

I second this. The Public IP addresses need to be created with Zone Redundancy if an Zone Redundant Sku size is selected.

**Comment by luke-taylor** [comment]
Thanks for raising this, going forward AVM will be zone redundant by default across all resources, this will be a good start.

**Comment by mafiaboy1994** [comment]
has this one been completed as i'm suffering from a similar issue when trying to deploy vpn GW AZ SKUs. The Public IP deployed is defaulting to a basic SKU and dynamically assigned IP despite using the latest version of this module?

**Comment by luke-taylor** [comment]
@mafiaboy1994 Hey, I have just cut the latest release, feature should be available as of v0.3.0, please read the release notes before upgrading.

**Comment by luke-taylor** [comment]
On further reflection, I believe this issue has convoluted two different issues. Rightly, the module was lacking the functionality to create zones for the Public IPs. However, in terms of the error, the zones on the PIPs are non-issue. 

Edge Zone refers to the extended locations with certain regions (typically metros) to provide lower latency (https://learn.microsoft.com/en-us/azure/public-multi-access-edge-compute-mec/key-concepts#parent-azure-regions). 

As far as the gateway is concerned, using a "AZ" SKU, suffices for zone redundancy in regions that cater to it. 

Having read the description for the variable name `edge_zone` I can understand the potential confusion, I will update the description in the next PR.
